components/content_settings/core/browser/brave_content_settings_pref_provider.cc

```diff
@@ -0,0 +1,61 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/content_settings/core/browser/brave_content_settings_pref_provider.h"
+
+#include "base/bind.h"
+#include "components/content_settings/core/browser/content_settings_pref.h"
+#include "components/content_settings/core/browser/website_settings_registry.h"
+
+namespace content_settings {
+
+BravePrefProvider::BravePrefProvider(PrefService* prefs,
+                                     bool incognito,
+                                     bool store_last_modified)
+    : PrefProvider(prefs, incognito, store_last_modified) {
+  brave_pref_change_registrar_.Init(prefs_);
+
+  WebsiteSettingsRegistry* website_settings =
+      WebsiteSettingsRegistry::GetInstance();
+  // Makes BravePrefProvder handle plugin type.
+  for (const WebsiteSettingsInfo* info : *website_settings) {
+    if (info->type() == CONTENT_SETTINGS_TYPE_PLUGINS) {
+      content_settings_prefs_.insert(std::make_pair(
+          info->type(),
+          std::make_unique<ContentSettingsPref>(
+              info->type(), prefs_, &brave_pref_change_registrar_,
+              info->pref_name(),
+              is_incognito_,
+              base::Bind(&PrefProvider::Notify, base::Unretained(this)))));
+      return;
+    }
+  }
+}
+
+void BravePrefProvider::ShutdownOnUIThread() {
+  brave_pref_change_registrar_.RemoveAll();
+  PrefProvider::ShutdownOnUIThread();
+}
+
+bool BravePrefProvider::SetWebsiteSetting(
+    const ContentSettingsPattern& primary_pattern,
+    const ContentSettingsPattern& secondary_pattern,
+    ContentSettingsType content_type,
+    const ResourceIdentifier& resource_identifier,
+    base::Value* in_value) {
+  // Flash type shouldn't be handled here. Its id is empty.
+  // It's type is plugin and id is empty string.
```

Comment:
I think this comment has some redundancy with "Its id is empty" and " id is empty string" making the reader think there are 2 ids, but I think there is only one.

Reply:
Updated.